### PR TITLE
[DEV-1929]: Allow filtering of notifications

### PR DESCRIPTION
**Pull request by sumanyumuku98**
# Description

Allow filtering of notifications based on `problemType` and `entityType`.

JIRA: [DEV-1929]

[DEV-1929]: https://causely.atlassian.net/browse/DEV-1929?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by cmiller-causely — APPROVED** [review]

**Comment by cmiller-causely** [comment]
Looks good to me.  It would be good if Endre reviews also.

File: README.md [hunk 1]
```diff
@@ -54,4 +54,17 @@ Payload fields:
...
 - `entityId`: The unique identifier for the entity involved.
-- `objectId`: A unique identifier for the specific object associated with this event, in this case it's the defectID.
\ No newline at end of file
+- `objectId`: A unique identifier for the specific object associated with this event, in this case it's the defectID.
+
+## Filtering Notifications
+A very simple filter configuration has been provided to filter notifications based on `problemType` and `entityType`. This can be set by customizing the `filterconfig` map as seen below:
+
+```yaml
+filterConfig:
```
Comment (esara) [inline README.md:63]: so the question is if you specify two types of entities (like container and application), does malfunction apply to both or only certain combination?

Comment (sumanyumuku98) [reply README.md:63]: To Both so for example if the config was `entityTypes`: [Container, Application] and `problemTypes`: [Malfunction] then the only allowed notifications would be `(Container, Malfunction)` and `(Application, Malfunction)`

Comment (esara) [reply README.md:63]: if that means every possible combination of entities and problems listed, that may not be specific enough for the user

Comment (cmiller-causely) [reply README.md:63]: I was also thinking about this.  I think a list of pairs Pod - Malfunction, Container - CPUCongestion ... is more useful to the user.  That being said, we can easily adjust this.  This is intended to give the user a starting point for filtering.

Comment (sumanyumuku98) [reply README.md:63]: Addressed in the latest commit!

**Pull request merged**